File: voiceping-sdk/src/main/java/com/smartwalkie/voicepingsdk/listeners/AudioRecorder.java
package com.smartwalkie.voicepingsdk.listeners;

/**
 * Interface for Recorder.
 */
public interface AudioRecorder {

    /**
     * Add interceptor that will be processed before sending the recorded audio data to server.
     *
     * @param audioInterceptor
     */
    void addAudioInterceptor(AudioInterceptor audioInterceptor);
}

File: voiceping-sdk/src/main/java/com/smartwalkie/voicepingsdk/listener/AudioRecorder.kt
package com.smartwalkie.voicepingsdk.listener

import com.smartwalkie.voicepingsdk.model.Channel

/**
 * Interface for Recorder.
 */
interface AudioRecorder {
    /**
     * Set interceptor that will be processed before the recorded audio data being encoded using
     * internal codec.
     *
     * @param audioInterceptor AudioInterceptor
     */
    fun setInterceptorBeforeEncoded(audioInterceptor: AudioInterceptor?)

    /**
     * Set interceptor that will be processed after the recorded audio data being encoded using
     * internal codec, but before sending them to server.
     *
     * @param audioInterceptor AudioInterceptor
     */
    fun setInterceptorAfterEncoded(audioInterceptor: AudioInterceptor?)

    val channel: Channel?

    val audioSessionId: Int
}